metrics/prometheus/prometheus.go

```diff
@@ -111,6 +111,29 @@ func (g prometheusGauge) Add(delta float64) {
 	g.GaugeVec.With(prometheus.Labels(g.Pairs)).Add(delta)
 }
 
+// RegisterCallbackGauge registers a Gauge with Prometheus whose value is
+// determined at collect time by the passed callback function. The callback
+// determines the value, and fields are ignored, so RegisterCallbackGauge
+// returns nothing.
+func RegisterCallbackGauge(namespace, subsystem, name, help string, callback func() float64) {
+	RegisterCallbackGaugeWithLabels(namespace, subsystem, name, help, prometheus.Labels{}, callback)
+}
+
+// RegisterCallbackGaugeWithLabels is the same as RegisterCallbackGauge, but
+// attaches a set of const label pairs to the metric.
+func RegisterCallbackGaugeWithLabels(namespace, subsystem, name, help string, constLabels prometheus.Labels, callback func() float64) {
```

Comment:
@bensigelman Ah sorry, didn't see that you already replied. The callback (`Collect` method) returns a bunch of [prometheus.Metric](http://godoc.org/github.com/prometheus/client_golang/prometheus#Metric) objects which in these kinds of collector callbacks are frequently constructed via the `MustNewConstMetric()` function, like in this example collector: https://github.com/prometheus/client_golang/blob/master/prometheus/expvar.go#L76

So the label values are returned as a string slice whose order has to match the label names that were specified at registration time.